File: spec/features/like_spec.rb
RSpec.describe "Likes" do
  let(:current_user) { create(:user) }

  context "when I already liked a post" do
    let!(:post) { create(:post, liking_users: [current_user]) }

    before(:each) do
      login(current_user)
      visit(posts_path)
    end

    it "shows a heart icon on the post" do
      expect(find("#post_#{post.id}")).to have_selector("a[title='Remove Like']")
    end

    it "lets me remove the like" do
      find("a[title='Remove Like']").click
      expect(find("#post_#{post.id}")).to have_selector("a[title='Like Post']")
    end
  end
end

File: spec/cypress/integration/like_spec.js
describe('Likes', function() {
    beforeEach(function() {
        cy.appScenario('basic_user').as('user');
        cy.factory('user').as('author');
    })

    context('when I already liked a post', function() {
        beforeEach(function() {
            cy.factory('post', [], { author_identifier: this.author.gid, liking_users_identifiers: [this.user.gid] }).as('post');
            cy.login(this.user);
        })

        it('shows a heart icon on the post', function() {
            cy.get(`#post_${this.post.id}`).get('a[title="Remove Like"]').should('exist');
        });

        it('lets me remove the like', function() {
            cy.get("a[title='Remove Like']").click();
            cy.get(`#post_${this.post.id}`).get("a[title='Like Post']").should('exist');
        })
    })
})
